src/pathfinding_service/model/token_network.py

```diff
@@ -56,6 +64,13 @@ def is_valid(self) -> bool:
             ratio = edge["view"].settle_timeout / edge["view"].reveal_timeout
             if ratio < DEFAULT_SETTLE_TO_REVEAL_TIMEOUT_RATIO:
                 return False
+
+        # check node reachabilities
+        for node in self.nodes:
+            node_reachability = self.address_to_reachability.get(node, AddressReachability.UNKNOWN)
+            if node_reachability != AddressReachability.REACHABLE:
+                return False
```

Comment:
When we start caring about performance, we should exclude unreachable nodes from the graph instead of removing all routes including them from the results. The same is true fro the `settle_timeout / reveal_timeout >= default ratio` condition.

That's harder for the capacity check, as the exact required capacity depends on the route.

Reply:
Sure, but then we have to keep a general graph with all edges and a pruned one which is used for routing.